ExecHashMap.hpp

```diff
@@ -0,0 +1,89 @@
+#include <iostream>
+#include <iterator>
+#include <numeric>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include <nlohmann/json.hpp>
+
+typedef struct IOPair {
+  std::vector<std::string> first;
+  std::vector<std::string> second;
+} IOPair;
+
+typedef struct ExecHashMap {
+  struct VectorHasher {
+    int operator()(const std::vector<std::string> &V) const {
+      std::string all = std::accumulate(
+          V.begin(), V.end(), std::string(),
+          [&](std::string x, std::string y) { return x + "," + y; });
+      return std::hash<std::string>{}(all);
+    }
+  };
+
+  std::unordered_map<std::vector<std::string>,
+                     std::vector<std::vector<std::string>>, VectorHasher>
```

Comment:
Can you `typedef std::vector<std::string> SomeName` To make this def look better?